src/matrix.js

```diff
@@ -0,0 +1,173 @@
+      } else if (response.statusCode === 200) {
+        let info;
+        try {
+          let dims = sizeOf(body);
+          this.robot.logger.info(`Image has dimensions ${JSON.stringify(dims)}, size ${body.length}`);
+          if (dims.type === 'jpg') { dims.type = 'jpeg'; }
+          info = { mimetype: `image/${dims.type}`, h: dims.height, w: dims.width, size: body.length };
+          return this.client.uploadContent(body, {name: url, type: info.mimetype, rawResponse: false, onlyContentUri: true}).done(content_uri => {
+            return this.client.sendImageMessage(envelope.room, content_uri, info, url).catch(err => {
+              if (err.name === 'UnknownDeviceError') {
+                this.handleUnknownDevices(err);
+                return this.client.sendImageMessage(envelope.room, content_uri, info, url);
+              }
+            });
+          });
+        } catch (error1) {
+          error = error1;
+          this.robot.logger.info(error.message);
+          return this.send(envelope, ` ${url}`);
+        }
+      }
+    });
+  }
+
+  run() {
+    this.robot.logger.info(`Run ${this.robot.name}`);
+    let client = sdk.createClient(process.env.HUBOT_MATRIX_HOST_SERVER || 'https://matrix.org');
+    this.robot.matrixClient = client;
+    return client.login('m.login.password', {
+      user: this.robot.name,
+      password: process.env.HUBOT_MATRIX_PASSWORD
+    }, (err, data) => {
+        if (err) {
+            this.robot.logger.error(err);
+            return;
+        }
+        this.user_id = data.user_id;
+        this.access_token = data.access_token;
+        this.device_id = data.device_id;
+        this.robot.logger.info(`Logged in ${this.user_id} on device ${this.device_id}`);
+        this.client = sdk.createClient({
+            baseUrl: process.env.HUBOT_MATRIX_HOST_SERVER || 'https://matrix.org',
+            accessToken: this.access_token,
+            userId: this.user_id,
+            deviceId: this.device_id,
+            sessionStore: new sdk.WebStorageSessionStore(localStorage)
+        });
+        this.client.on('sync', (state, prevState, data) => {
+            switch (state) {
+              case "PREPARED":
+                this.robot.logger.info(`Synced ${this.client.getRooms().length} rooms`);
+                return this.emit('connected');
+            }
+        });
+        this.client.on('Room.timeline', (event, room, toStartOfTimeline) => {
+            if ((event.getType() === 'm.room.message') && (toStartOfTimeline === false)) {
+                this.client.setPresence("online");
+                let message = event.getContent();
+                let name = event.getSender();
+                let prettyname = room.currentState._userIdsToDisplayNames[name];
+                let user = this.robot.brain.userForId(name, { name: prettyname });
+                user.room = room.roomId;
+                if (user.name !== this.user_id) {
```

Comment:
good catch, I just updated this and also gave it name flexibility so it's displayname/botname can be different from it's matrix id